Unable to connect a cloud

Hi,
I'm currently trying to get the yet-another-docker-plugin working with the latest Jenkins version in a docker container. So basically I'm setting up my Jenkins instance and trying to add the host as a Cloud.

Here is the Jenkins setup (very simple): https://github.com/foxylion-playground/jenkins-vagrant-docker

The unix socket is mounted into the running jenkins container (and docker is working inside). When adding `unix:///var/run/docker.socket` I get the following errors when testing the connection:

Using NETTY:
```
javax.servlet.ServletException: java.lang.NoClassDefFoundError: Could not initialize class com.github.kostyasha.yad_docker_java.io.netty.channel.epoll.IovArray
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:796)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:876)
	at org.kohsuke.stapler.MetaClass$5.doDispatch(MetaClass.java:233)
	at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:58)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:746)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:876)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:649)
	at org.kohsuke.stapler.Stapler.service(Stapler.java:238)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:812)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1669)
	at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:135)
	at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:126)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:49)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)
	at hudson.security.UnwrapSecurityExceptionFilter.doFilter(UnwrapSecurityExceptionFilter.java:51)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at jenkins.security.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:117)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.providers.anonymous.AnonymousProcessingFilter.doFilter(AnonymousProcessingFilter.java:125)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.ui.rememberme.RememberMeProcessingFilter.doFilter(RememberMeProcessingFilter.java:142)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.ui.AbstractProcessingFilter.doFilter(AbstractProcessingFilter.java:271)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at jenkins.security.BasicHeaderProcessor.doFilter(BasicHeaderProcessor.java:93)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessionContextIntegrationFilter.java:249)
	at hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContextIntegrationFilter2.java:67)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:76)
	at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:171)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:49)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:82)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.kohsuke.stapler.DiagnosticThreadNameFilter.doFilter(DiagnosticThreadNameFilter.java:30)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:553)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
	at org.eclipse.jetty.server.Server.handle(Server.java:499)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:311)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:544)
	at winstone.BoundedExecutorService$1.run(BoundedExecutorService.java:77)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.github.kostyasha.yad_docker_java.io.netty.channel.epoll.IovArray
	at com.github.kostyasha.yad_docker_java.io.netty.channel.epoll.EpollEventLoop.<init>(EpollEventLoop.java:62)
	at com.github.kostyasha.yad_docker_java.io.netty.channel.epoll.EpollEventLoopGroup.newChild(EpollEventLoopGroup.java:130)
	at com.github.kostyasha.yad_docker_java.io.netty.channel.epoll.EpollEventLoopGroup.newChild(EpollEventLoopGroup.java:35)
	at com.github.kostyasha.yad_docker_java.io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:84)
	at com.github.kostyasha.yad_docker_java.io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:58)
	at com.github.kostyasha.yad_docker_java.io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:47)
	at com.github.kostyasha.yad_docker_java.io.netty.channel.MultithreadEventLoopGroup.<init>(MultithreadEventLoopGroup.java:58)
	at com.github.kostyasha.yad_docker_java.io.netty.channel.epoll.EpollEventLoopGroup.<init>(EpollEventLoopGroup.java:100)
	at com.github.kostyasha.yad_docker_java.io.netty.channel.epoll.EpollEventLoopGroup.<init>(EpollEventLoopGroup.java:87)
	at com.github.kostyasha.yad_docker_java.io.netty.channel.epoll.EpollEventLoopGroup.<init>(EpollEventLoopGroup.java:64)
	at com.github.kostyasha.yad_docker_java.com.github.dockerjava.netty.NettyDockerCmdExecFactory$UnixDomainSocketInitializer.init(NettyDockerCmdExecFactory.java:220)
	at com.github.kostyasha.yad_docker_java.com.github.dockerjava.netty.NettyDockerCmdExecFactory.init(NettyDockerCmdExecFactory.java:196)
	at com.github.kostyasha.yad_docker_java.com.github.dockerjava.core.DockerClientImpl.withDockerCmdExecFactory(DockerClientImpl.java:161)
	at com.github.kostyasha.yad.client.ClientBuilderForConnector.build(ClientBuilderForConnector.java:185)
	at com.github.kostyasha.yad.DockerConnector$DescriptorImpl.doTestConnection(DockerConnector.java:211)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.kohsuke.stapler.Function$InstanceFunction.invoke(Function.java:324)
	at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:167)
	at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:100)
	at org.kohsuke.stapler.MetaClass$1.doDispatch(MetaClass.java:124)
	at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:58)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:746)
	... 58 more
```

Using JERSEY:
```
com.github.kostyasha.yad_docker_java.org.newsclub.net.unix.NativeUnixSocket.connect(Ljava/lang/String;Ljava/io/FileDescriptor;)V
com.github.kostyasha.yad_docker_java.javax.ws.rs.ProcessingException: com.github.kostyasha.yad_docker_java.org.newsclub.net.unix.NativeUnixSocket.connect(Ljava/lang/String;Ljava/io/FileDescriptor;)V
	at com.github.kostyasha.yad_docker_java.org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:261)
	at com.github.kostyasha.yad_docker_java.org.glassfish.jersey.client.JerseyInvocation$2.call(JerseyInvocation.java:701)
	at com.github.kostyasha.yad_docker_java.org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at com.github.kostyasha.yad_docker_java.org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at com.github.kostyasha.yad_docker_java.org.glassfish.jersey.internal.Errors.process(Errors.java:228)
	at com.github.kostyasha.yad_docker_java.org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:444)
	at com.github.kostyasha.yad_docker_java.org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:697)
	at com.github.kostyasha.yad_docker_java.org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:420)
	at com.github.kostyasha.yad_docker_java.org.glassfish.jersey.client.JerseyInvocation$Builder.get(JerseyInvocation.java:316)
	at com.github.kostyasha.yad_docker_java.com.github.dockerjava.jaxrs.VersionCmdExec.execute(VersionCmdExec.java:26)
	at com.github.kostyasha.yad_docker_java.com.github.dockerjava.jaxrs.VersionCmdExec.execute(VersionCmdExec.java:13)
	at com.github.kostyasha.yad_docker_java.com.github.dockerjava.jaxrs.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:23)
	at com.github.kostyasha.yad_docker_java.com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
	at com.github.kostyasha.yad.DockerConnector$DescriptorImpl.doTestConnection(DockerConnector.java:213)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.kohsuke.stapler.Function$InstanceFunction.invoke(Function.java:324)
	at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:167)
	at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:100)
	at org.kohsuke.stapler.MetaClass$1.doDispatch(MetaClass.java:124)
	at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:58)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:746)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:876)
	at org.kohsuke.stapler.MetaClass$5.doDispatch(MetaClass.java:233)
	at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:58)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:746)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:876)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:649)
	at org.kohsuke.stapler.Stapler.service(Stapler.java:238)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:812)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1669)
	at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:135)
	at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:126)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:49)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)
	at hudson.security.UnwrapSecurityExceptionFilter.doFilter(UnwrapSecurityExceptionFilter.java:51)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at jenkins.security.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:117)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.providers.anonymous.AnonymousProcessingFilter.doFilter(AnonymousProcessingFilter.java:125)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.ui.rememberme.RememberMeProcessingFilter.doFilter(RememberMeProcessingFilter.java:142)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.ui.AbstractProcessingFilter.doFilter(AbstractProcessingFilter.java:271)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at jenkins.security.BasicHeaderProcessor.doFilter(BasicHeaderProcessor.java:93)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessionContextIntegrationFilter.java:249)
	at hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContextIntegrationFilter2.java:67)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:76)
	at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:171)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:49)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:82)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.kohsuke.stapler.DiagnosticThreadNameFilter.doFilter(DiagnosticThreadNameFilter.java:30)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:553)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
	at org.eclipse.jetty.server.Server.handle(Server.java:499)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:311)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:544)
	at winstone.BoundedExecutorService$1.run(BoundedExecutorService.java:77)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.UnsatisfiedLinkError: com.github.kostyasha.yad_docker_java.org.newsclub.net.unix.NativeUnixSocket.connect(Ljava/lang/String;Ljava/io/FileDescriptor;)V
	at com.github.kostyasha.yad_docker_java.org.newsclub.net.unix.NativeUnixSocket.connect(Native Method)
	at com.github.kostyasha.yad_docker_java.org.newsclub.net.unix.AFUNIXSocketImpl.connect(AFUNIXSocketImpl.java:134)
	at com.github.kostyasha.yad_docker_java.org.newsclub.net.unix.AFUNIXSocket.connect(AFUNIXSocket.java:97)
	at com.github.kostyasha.yad_docker_java.com.github.dockerjava.jaxrs.ApacheUnixSocket.connect(ApacheUnixSocket.java:62)
	at com.github.kostyasha.yad_docker_java.com.github.dockerjava.jaxrs.UnixConnectionSocketFactory.connectSocket(UnixConnectionSocketFactory.java:74)
	at com.github.kostyasha.yad_docker_java.org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:134)
	at com.github.kostyasha.yad_docker_java.org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:353)
	at com.github.kostyasha.yad_docker_java.org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:380)
	at com.github.kostyasha.yad_docker_java.org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at com.github.kostyasha.yad_docker_java.org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:184)
	at com.github.kostyasha.yad_docker_java.org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:88)
	at com.github.kostyasha.yad_docker_java.org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at com.github.kostyasha.yad_docker_java.org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184)
	at com.github.kostyasha.yad_docker_java.org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:71)
	at com.github.kostyasha.yad_docker_java.org.glassfish.jersey.apache.connector.ApacheConnector.apply(ApacheConnector.java:435)
	at com.github.kostyasha.yad_docker_java.org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:252)
	... 81 more
```

Sorry to bother you with that question, but I couldn't find any solution to that. :disappointed: 

It's btw pretty easy to try it yourself:
```
git clone git@github.com:foxylion-playground/jenkins-vagrant-docker.git
cd jenkins-vagrant-docker
vagrant up
```
Open http://10.111.0.10/configure and login with admin/admin. Then add the "Yet Another Docker Plugin" cloud with `unix:///var/run/docker.sock` and click "Test connection".

--- a/docs/FEATURES.adoc
+++ b/docs/FEATURES.adoc
@@ -1,6 +1,9 @@
 == Jenkins Cloud Slaves provisioning
 
 * TLS (RSA,ECDSA) based credentials connection to docker daemon.
+* Connection to docker daemon:
+** tcp or socket (socket works only for JERSEY impl)
+** JERSEY or NETTY implementations (provided by docker-java)
 * Slave image pulling strategies.
 * Exact configuration for container lifecycle:
 ** pull

--- a/docs/FEATURES.adoc
+++ b/docs/FEATURES.adoc
@@ -15,5 +15,6 @@
 ** JNLP (requires working access from slave to master)
 ** SSH (requires working access from master to slave)
 * Docker fingerprints badges
-* Full shading of docker-java library (doesn't conflict with docker-traceability-plugin or any other plugin using docker-java)
+* Full shading of docker-java library (doesn't conflict with docker-traceability-plugin or any other plugin
+using docker-java). (May have issues because of JNI since 0.1.0-rc24.)
 * Faster node provisioning without delay

--- a/yet-another-docker-java/pom.xml
+++ b/yet-another-docker-java/pom.xml
@@ -215,10 +215,10 @@
                             <pattern>de.gesellix.socketfactory.unix</pattern>
                             <shadedPattern>${shade-prefix}.de.gesellix.socketfactory.unix</shadedPattern>
                         </relocation>
-                        <relocation>
-                            <pattern>org.newsclub</pattern>
-                            <shadedPattern>${shade-prefix}.org.newsclub</shadedPattern>
-                        </relocation>
+                        <!--<relocation>-->
+                            <!--<pattern>org.newsclub</pattern>-->
+                            <!--<shadedPattern>${shade-prefix}.org.newsclub</shadedPattern>-->
+                        <!--</relocation>-->
                         <relocation>
                             <pattern>socketfactory.spi</pattern>
                             <shadedPattern>${shade-prefix}.socketfactory.spi</shadedPattern>

--- a/yet-another-docker-java/pom.xml
+++ b/yet-another-docker-java/pom.xml
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
     <!-- No parent, otherwise we inherit wrong versions from transient jenkins dependencies. -->
@@ -63,7 +64,7 @@
         <dependency>
             <groupId>com.github.docker-java</groupId>
             <artifactId>docker-java</artifactId>
-            <version>3.0.5</version>
+            <version>3.0.6</version>
         </dependency>
         <dependency>
             <groupId>org.slf4j</groupId>
@@ -215,12 +216,17 @@
                             <shadedPattern>${shade-prefix}.de.gesellix.socketfactory.unix</shadedPattern>
                         </relocation>
                         <relocation>
-                            <pattern>org.newsclub.net.unix</pattern>
-                            <shadedPattern>${shade-prefix}.org.newsclub.net.unix</shadedPattern>
+                            <pattern>org.newsclub</pattern>
+                            <shadedPattern>${shade-prefix}.org.newsclub</shadedPattern>
                         </relocation>
                         <relocation>
                             <pattern>socketfactory.spi</pattern>
                             <shadedPattern>${shade-prefix}.socketfactory.spi</shadedPattern>
+                        </relocation>
+                        <!-- unix socket -->
+                        <relocation>
+                            <pattern>org.scijava</pattern>
+                            <shadedPattern>${shade-prefix}.org.scijava</shadedPattern>
                         </relocation>
 
                         <!-- docker-java forked jersey -->

